rwengine/src/engine/GameWorld.cpp

```diff
@@ -459,6 +459,15 @@ GarageInfo* GameWorld::createGarage(const glm::vec3 coord0,
     return info;
 }
 
+Payphone* GameWorld::createPayphone(const glm::vec2 coord) {
+    int id = payphones.size();
+    Payphone* payphone = new Payphone(this, id, coord);
```

Comment:
Memory leak?

Reply:
Maybe it's better to keep in vector unique_ptrs?

Reply:
But it's not uniquely-owned

Reply:
You can create unique_ptr with this memory when inserting into vector. Other way is usage of `std::move`.

Reply:
But we still need to return back valid object after inserting into vector, and it's no longer valid after ```std::move``` Or I guess you can just do ```back``` on vector to return newly created object.

Reply:
That's all nice, but it doesn't really go well with all this ScriptObjectType stuff

Reply:
Raw ptr to memory allocated by unique_ptr you can get using `get`.

Reply:
Doesn't that break the whole purpose of having a unique_ptr then?

Reply:
Nope, there's only one owner of memory(so overhead as raw ptr) and you have quarantine that memory is freed at the end. 

Good habit is to use ref instead of raw ptr when borrowing memory. As I suggested on IRC lately.